### ISSUE TITLE
String type

### ISSUE THREAD

**Issue by imkiva** [covscript/drafts#2]
String is a **first-class** type.

Because we have CNI, so our decision will be one of:
* std::string
* std::u32string

**Comment by mikecovlee** [comment]
No, don't use `std::u32string` in parser which have low space efficiency.
`std::istream` -> `std::string` -> lexer(`std::u32string` -> lex -> `std::string`) -> parser

**Comment by imkiva** [comment]
> No, don't use `std::u32string` in parser which have low space efficiency.
> `std::istream` -> `std::string` -> lexer(`std::u32string` -> lex -> `std::string`) -> parser

Yes, this is what I am talking about.

**Comment by imkiva** [comment]
I prefer:
* VM manages strings as `std::string`
* User uses strings as `std::u32string`
* We hide the automated conversion between the user and the VM.

**Comment by ice1000** [comment]
Ok, just saying.

**Comment by mikecovlee** [comment]
Do not use u32string in vm, whether CNI will be a nightmare.

**Comment by imkiva** [comment]
> Do not use u32string in vm, whether CNI will be a nightmare.

VM only performs the conversation
